scripts/html_to_docbook.py

```diff
@@ -0,0 +1,113 @@
+import json
+from pathlib import Path
+import logging as log
+import os
+import re
+
+import lib.html_normalizer as html_norm
+import lib.image_localizer as image_loc
+import lib.herold as herold
+import lib.docbook_template as dbtemplate
+
+log.basicConfig(format="%(asctime)s %(levelname)s %(message)s", level=log.DEBUG)
+
+BASE_P = Path("lesswrong.com/Rationality: A-Z")
+
+BOOK_PATH = Path("lesswrong.com/book.english")
+
+def collect_persons(book_info: dict, meta: dict) -> None:
+    book_info.setdefault("user", {})
+    book_info.setdefault("reviewer", {})
+    
+    if meta.get("user"):
+      book_info["user"][meta["user"]] = book_info["user"].setdefault(meta["user"], 0) + 1
+
+    if meta.get("reviewer"):
+      book_info["reviewer"][meta["reviewer"]] = 1
+      
+      
+def set_publication_date(book_info: dict, meta: dict) ->  None:
+    if meta["modified_at"] is None:
+        return
+
+    if book_info.get("modified_at") is None or book_info["modified_at"] < meta["modified_at"]:
+        book_info["modified_at"] = meta["modified_at"]
+      
+      
+def get_authors(book_info: dict) -> list:
+    return sorted(
+        book_info["user"].keys(),
+        key=lambda x: book_info["user"][x],
+        reverse=True,
+    )
+    
+    
+def get_reviewers(book_info: dict) -> list:
+    return list(filter(
+        lambda x: not book_info["user"].get(x),
+        book_info["reviewer"]
+    ))
+    
+def title_to_filename(title: str) -> str:
+    title = re.sub('[^\w\d\s]', '', title)
+    title = re.sub('^\s+|\s+$', '', title)
+    title = re.sub('\s+', '_', title)
```

Comment:
Nit: for regexes it's safer to use raw strings - `re.sub(r'\s+', '_', title)`, also for performance improvement (not our case, but it's already marked as nitpick) regexes can be compiled once as global variables/constants or function parameters with default value (I prefer this way although can look a bit ugly in this case):
```
def title_to_filename(title: str, not_allowed_regex=re.compile("[^\w\d\s]"), spaces_regex=...) -> str:
    title = not_allowed_regex.sub("". title.strip())
    title = spaces_regex.sub("_", title)
    return title
```
and good to run black against code